editoast/src/views/v2/train_schedule.rs

```diff
@@ -318,75 +370,252 @@ pub struct MrspPoint {
-            times: vec![27.5, 35.5],
+            times: vec![27500, 35500],
             energy_consumption: 100.0,
         },
         provisional: ReportTrain {
             speeds: vec![10.0, 27.0],
             positions: vec![20, 50],
-            times: vec![27.5, 35.5],
+            times: vec![27500, 35500],
             energy_consumption: 100.0,
         },
         final_output: CompleteReportTrain {
             report_train: ReportTrain {
                 speeds: vec![10.0, 27.0],
                 positions: vec![2000, 5000],
-                times: vec![27.5, 35.5],
+                times: vec![27500, 35500],
                 energy_consumption: 100.0,
             },
             signal_sightings: vec![SignalSighting {
                 signal: "signal.0".into(),
-                time: 28.0,
+                time: 28000,
                 position: 3000,
                 state: "VL".into(),
             }],
             zone_updates: vec![ZoneUpdate {
                 zone: "zone.0".into(),
-                time: 28.0,
+                time: 28000,
                 position: 3000,
                 is_entry: true,
             }],
             spacing_requirements: vec![SpacingRequirement {
                 zone: "zone.0".into(),
-                begin_time: 30.0,
-                end_time: 35.5,
+                begin_time: 30000,
+                end_time: 35500,
             }],
             routing_requirements: vec![RoutingRequirement {
                 route: "route.0".into(),
-                begin_time: 32.0,
+                begin_time: 32000,
                 zones: vec![],
             }],
         },
         mrsp: Mrsp(vec![]),
         power_restriction: "".into(),
-    }))
+    };
+    let res = SimulationResult::Success(Box::new(sim_success));
+
+    // Cache the simulation response
+    redis_conn
+        .json_set_ex(&hash, &res, CACHE_SIMULATION_EXPIRATION)
+        .await?;
+
+    // Return the response
+    Ok(res)
 }
 
+async fn build_simulation_request(
+    conn: &mut PgConnection,
+    train_schedule: &TrainSchedule,
+    path_items_position: &[u64],
+    path: SimulationPath,
+) -> Result<SimulationRequest> {
+    // Get rolling stock
+    let rolling_stock_name = train_schedule.rolling_stock_name.clone();
+    let rolling_stock = RollingStockModel::retrieve(conn, rolling_stock_name.clone())
+        .await?
+        .expect("Rolling stock should exist since the pathfinding succeeded");
+    // Get electrical_profile_set_id
+    let timetable_id = train_schedule.timetable_id;
+    let timetable = Timetable::retrieve(conn, timetable_id)
+        .await?
+        .expect("Timetable should exist since it's a foreign key");
+
+    assert_eq!(path_items_position.len(), train_schedule.path.len());
+
+    // Project path items to path offset
+    let path_items_to_position: HashMap<_, _> = train_schedule
+        .path
+        .iter()
+        .map(|p| &p.id)
+        .zip(path_items_position.iter())
```

Comment:
It's probably not really that useful to have `&u64` as values, let's make it `u64` directly.

```suggestion
        .zip(path_items_position.iter().copied())
```